src/components/database/Queries.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import {database} from '../../firebaseFeatures.js';
+import type {UserType, PendingType, ActionType} from '../admin/FlowTypes.js';
+
+function sanitize(collection: string,
+    user: any): UserType | ActionType | PendingType {
+  if (collection === 'active-members') {
+    return ({
+      adminNote: user.adminNote,
+      email: user.email,
+      isAdmin: user.isAdmin,
+      joinDate: user.joinDate.toDate().toLocaleString(),
+      name: user.name,
+      needsAttention: user.needsAttention,
+      partnerEmail: user.partnerEmail,
+      count: user.count,
+      id: user.id,
+    }: UserType);
+  } else if (collection === 'pending-members') {
+    return ({
+      count: user.count,
+      email: user.email,
+      date: user.date.toDate().toLocaleString(),
+      partnerEmail: user.partnerEmail,
+    }: PendingType);
+  } else if (collection === 'actions') {
+    return ({
+      count: user.count,
+      date: user.date.toDate().toLocaleString(),
+      message: user.message,
+    }: ActionType);
+  } else {
+    throw Error('The collection you asked for doesn\'t exist.');
+  }
+}
+
+function dataSanitization(data: any,
+    collection: string): Array<UserType | ActionType | PendingType> {
+  const result = [];
+  data.forEach((doc: any) => {
+    result.push(sanitize(collection, doc.data()));
+  });
+
+  return result;
```

Comment:
```suggestion
  return data.map((doc: any) => doc.data())
    .map((data:any) => sanitize(collection, data));
```

If you feel very fancy:

```suggestion
  const sanitizeOnCollection = (data:any) => sanitize(collection, data);
  return data.map((doc: any) => doc.data())
    .map(sanitizeOnCollection);
```

Reply:
I wanted to use `.map()` instead of `.forEach()` initially, but apparently, firestore snapshots don't support the `.map()` function. However, after a quick search, I've discovered that if we want to use `map`, we have to call it like this: `data.docs.map(...)`. https://stackoverflow.com/questions/46614055/map-items-of-collection-snapshot-in-firebase-firestore/46614056

Reply:
ah, sorry I didn't know this. `data.docs.map` sounds good :-)